src/test/java/com/onfido/integration/TestBase.java

```diff
@@ -142,4 +147,75 @@ public String getTaskIdByPartialId(UUID workflowRunId, String partialId) throws
         .get(0)
         .getId();
   }
+
+  public WatchlistMonitor createWatchlistMonitor(
+      UUID applicantId, WatchlistMonitorBuilder.ReportNameEnum reportName) throws ApiException {
+    return onfido.createWatchlistMonitor(
+        new WatchlistMonitorBuilder().applicantId(applicantId).reportName(reportName));
+  }
+
+  private Method getMethod(String methodName, Object[] params) throws NoSuchMethodException {
+    Class<?>[] paramTypes = new Class<?>[params.length];
+    for (int i = 0; i < params.length; i++) {
+      paramTypes[i] = params[i].getClass();
+    }
+
+    return onfido.getClass().getMethod(methodName, paramTypes);
+  }
+
+  public Object repeatRequestUntilStatusChanges(
+      String methodName, Object[] params, Object status, int maxRetries, int sleepTime)
+      throws NoSuchMethodException,
+          IllegalAccessException,
+          InterruptedException,
+          InvocationTargetException {
+    Method method = getMethod(methodName, params);
+    Object instance = method.invoke(onfido, params);
+    boolean isInstanceOfReport = instance instanceof Report;
+
+    if (isInstanceOfReport) {
+      instance = ((Report) instance).getActualInstance();
+    }
```

Comment:
I think it would be better moving the call to `getActualInstance()` outside this function to also validate methods like `getId()` and `getName()` on the generic report object,.

Reply:
I use `repeatRequestUntilStatusChanges` for workflow runs ([example](https://github.com/onfido/onfido-java/pull/114/files/6dfd08e0f37798666c999e5af26ddf2efac44bac#diff-b1f8871a5a9e28b0cef94712860f3b10d5e546fccd942fb78ac5dd143435d290R77)) which doesn't have `.getActualInstance()`. That's why I'm validating here if the object is an instance of Report.

Reply:
As discussed, it's fine to come back on this point in a subsequent PR.